src/NServiceBus.Core/Transports/ErrorContext.cs

```diff
@@ -35,7 +19,7 @@ public ErrorContext(Exception exception, Dictionary<string, string> headers, str
         /// <param name="transportTransaction">Transaction (along with connection if applicable) used to receive the message.</param>
         /// <param name="immediateProcessingFailures">Number of failed immediate processing attempts.</param>
         /// <param name="context">A <see cref="ContextBag" /> which can be used to extend the current object.</param>
-        public ErrorContext(Exception exception, Dictionary<string, string> headers, string transportMessageId, byte[] body, TransportTransaction transportTransaction, int immediateProcessingFailures, ReadOnlyContextBag context)
+        public ErrorContext(Exception exception, Dictionary<string, string> headers, string transportMessageId, byte[] body, TransportTransaction transportTransaction, int immediateProcessingFailures, ContextBag context)
```

Comment:
Why are we changing the type from `ReadOnlyContextBag` to `ContextBag`? The only use of the argument is an assignment to `Extensions`, which is a `ReadOnlyContextBag`.

Reply:
I'll revert for now, we'll need it to be `ContextBag` when the `OnComplete` is in but then it can make sense to add things to the context in `OnError` but I'll make that change in the other PR

Reply:
I wanted to make the exact same comment during the review but then I realized it is required to be a context bag to even be extendable at all. Why reverting since this was the intention of this change anyway? Or at least that's how I understood @andreasohlund original intention

Reply:
https://github.com/Particular/NServiceBus/pull/5936 will make it extendable again